pkg/dogstatsd/replay/writer.go

```diff
@@ -333,8 +338,10 @@ func (tc *TrafficCaptureWriter) StopCapture() {
 // Enqueue enqueues a capture buffer so it's written to file.
 func (tc *TrafficCaptureWriter) Enqueue(msg *CaptureBuffer) bool {
 	qd := false
+	tc.RLock()
+	defer tc.RUnlock()
```

Comment:
Does this lock have an impact on the performance? (I guess Enqueue is called multiple times).

Reply:
It was originally being called `IsOngoing()`, I've changed it to use an `atomic.Bool`. I need to do another refactor and tested again as ended up in a deadlock for some reason 😕

Reply:
Also, just realised what the deadlock I caused myself 😅